app/main.py

```diff
@@ -1,3 +1,18 @@
+cache_dict = {}
+
+
 def cache(func):
-    # Write your code here
-    pass
+    def inner(*args):
+        global cache_dict
+        if not cache_dict or func not in cache_dict:
+            cache_dict.update({func: {args: func(*args)}})
+            print("Calculating new result")
+            return func(*args)
```

Comment:
you have a duplicate `return`, take it outside `if-else` statement; try to rewrite your decorator not to have `return` duplicated